nsis/src/nsis/NsisLoader.java

```diff
@@ -144,26 +180,46 @@ public Data createData(Program program, Listing listing, Address address,
 		return null;
 	}
 
-	private void processBlockHeaders(Program program, TaskMonitor monitor,
-			BinaryReader reader, long nsis_header_offset) {
-		int block_header_offset = NsisScriptHeader.getHeaderSize();
+	/**
+	 * Initializes the block headers and adds them to the "Program Trees" view in
+	 * Ghidra.
+	 * 
+	 * @param program
+	 * @param reader
+	 * @param startingAddr, the Address where the nsis script header starts
+	 * @throws IOException
+	 * @throws AddressOverflowException
+	 * @throws MemoryConflictException
+	 * @throws DuplicateNameException
+	 * @throws LockException
+	 */
+	private void initBlockHeaders(Program program, BinaryReader reader, Address startingAddr,
+			FileBytes fileBytes) throws IOException, LockException, DuplicateNameException,
+			MemoryConflictException, AddressOverflowException {
+		int block_header_offset = 0;
+
+		Memory memory = program.getMemory();
+		MemoryBlock new_block = memory.createInitializedBlock(".block_headers", startingAddr,
+				fileBytes, 0, fileBytes.getSize(), false);
+
+		new_block.setRead(true);
+		new_block.setWrite(false);
+		new_block.setExecute(false);
+
 		for (int i = 0; i < NsisConstants.NB_NSIS_BLOCKS; i++) {
 			System.out.printf("Processing block at offset %08x\n",
-					block_header_offset);
-			ghidra.program.model.address.Address block_address;
-			block_address = program.getAddressFactory().getDefaultAddressSpace()
-					.getAddress(block_header_offset);
+					block_header_offset + startingAddr.getOffset());
+			Address block_address = startingAddr.add(block_header_offset);
+
+			reader.setPointerIndex(startingAddr.getOffset() + block_header_offset);
```

Comment:
Since `block_address` already is at the right place, is it possible to get the value calculated by `startingAddr.getOffset() + block_header_offset` from the `Address` object?